include/core/rtos/EventFlags.hpp

```diff
@@ -0,0 +1,122 @@
+
+#ifndef _EVT_RTOS_EVENTFLAGS_
+#define _EVT_RTOS_EVENTFLAGS_
+
+#include <core/rtos/Initializable.hpp>
+
+namespace core::rtos {
+
+/**
+ * Class that wraps a ThreadX Event Flags Group. Event flags are essentially
+ * a 32-bit mask, with each bit representing a different flag.\n\n
+ *
+ *
+ * This class extends Initializable, and like all other Initializable classes must be passed into the
+ * Threadx::startKernel() method as part of the initList parameter.
+ */
+class EventFlags : public Initializable {
+public:
+    /**
+     * Constructor for EventFlags.
+     *
+     * @param[in] name the name of the EventFlags.
+     */
+    EventFlags(char* name);
+
+    TXError init(core::rtos::BytePoolBase& pool) override;
+
+    /**
+     * Destructor for an EventFlags.
+     */
+    ~EventFlags();
+
+    /**
+     * Sets the eventFlags of the mask according to the setOption.
+     *
+     * @param[in] mask A bitmask describing what flags are to be interacted with.
+     * @param[in] clearNonMaskedFlags If true, all the flags not included in the mask will be set to 0.
```

Comment:
Yeah. I debated about this a bit. I don't think there's a succinct way to clearly describe that it's just an AND vs. OR with just the parameter name. I added a comment describing it, but I think clearNonMaskedFlags is pretty good because that's literally exactly what this parameter determines. It describes the high level behaviour of the AND vs OR operation.

Reply:
But if this is true, it won't just clear the non-masked flags. It also won't set the masked flags. Just saying this decides whether to clear non-masked flags is an insufficient description

Reply:
No, it always sets the masked flags. clearNonMaskedFlags just determines if it sets the masked flags AND clears the non masked flags, or it sets the masked flags and then just does nothing to the other flags.

Reply:
Looks to me like they do are just doing a bitwise and or or:
and: https://github.com/eclipse-threadx/threadx/blob/485a02faec6edccef14812ddce6844af1d7d2eef/common/src/tx_event_flags_set.c#L170-L171
or: https://github.com/eclipse-threadx/threadx/blob/485a02faec6edccef14812ddce6844af1d7d2eef/common/src/tx_event_flags_set.c#L191-L192

Am I misunderstanding something here?

Reply:
Yeah, that's correct. So if clearNonMaskedFlags is false, we do the bitwise OR. if clearNonMaskedFlags is true, we do the bitwise AND. The name describes the high-level functionality of those operations. 
I think what you may be confused about is that to me, the clearNonMaskedFlags is an extra functionality question. The function ALWAYS sets the masked flags, and clearNonMaskedFlags just specifies what to do with the flags not included in the mask.

Reply:
What I'm saying is that when `clearNonMaskedFlags` is true, this function _does not_ set the masked flags. It clears the nonmasked flags but leave the masked flags alone, even if they were 0 beforehand

Reply:
Okay so I talked it through with Matt, and we decided to split it into two methods,
set(mask) and clear(mask).
set(mask)    => eventFlags | mask (i.e. it sets the masked bits to 1 and keeps everything else the same)
clear(mask) => eventFlags & ~mask (i.e. it clears the masked bits and keeps everything else the same)
I think this is way clearer and doesn't lose out on any functionality, plus it means that someone can understand what is being done to the eventFlag by just glancing at the name of the method, instead of what arguments are being passed to it.